clang/lib/Headers/inttypes.h

```diff
@@ -18,6 +18,11 @@
 #error MSVC does not have inttypes.h prior to Visual Studio 2013
 #endif
 
+#ifndef __cplusplus
```

Comment:
Do we want this enabled even if the user compiles a C file with `-fno-checkedc-extension`?

Maybe we can create a macro like `__checkedc` which gets defined if Checked C is enabled. Then we can use that macro as guard here.

Reply:
Yes, we need to resolve this issue. We also need to consider the use case when inttypes_checked.h is explicitly included.

Reply:
+1 to creating a macro like `__checkedc`.  That would make it possible for the public headers of third-party libraries to automatically generate Checked C declarations when loaded by the Checked C compiler while remaining compatible with plain C compilers as well: in effect, supporting implicit inclusion of checked headers for third-party libraries, just as you are doing for the system headers in this PR.  (I was thinking about filing a separate issue for this.)

Reply:
Thanks! I have created the `__checkedc` macro in the latest commit.